### PR TITLE
Studio: Increase timeout in wp-cli-process and terminate database import on timeout

### PR DESCRIPTION
**Pull request by katinthehatsite**
## Related issues

Closes https://github.com/Automattic/dotcom-forge/issues/10032

## Proposed Changes

This PR increases the timeout for the `wp-cli-process` so that the database import does not fail for larger sites.

Some related details:

* While testing the backup from https://github.com/Automattic/dotcom-forge/issues/10032 , I noticed that the error is specifically related to executing wp-cli commands: `Warning during import of /var/folders/bj/1qgxs1qd6_z20tw9llkw842h0000gn/T/studio_backupsJr1QS/sql/wp_aiowps_events.sql: error when executing wp-cli command`. 
* After inspecting the error more and comparing other .sql files from the `wp_aiowps_` that have the same configuration, I noticed that `wp_aiowps_events.sql` is substantially larger.
* I suggest increasing the timeout in `wp-cli-process` so that the larger .sql files get processed without failing. 

## Testing Instructions

**To test the fix for importing larger .sql files**

* Download the backup for the site reported in CfT: https://mc.a8c.com/rewind/debugger.php?site=220627588# (you can produce the backup by clicking on `Create backup link` and download it after)
* Pull the changes from this branch
* Run `npm install && npm start`
* Open the Studio app
* Navigate to the Import / Export tab 
* Drop the backup into the Import field
* Wait for the import to finish
* Confirm the import finishes successfully
* You can also repeat the same steps on trunk and confirm that the database import fails

**To test the fix for terminating database import process when it fails**:

* Download the backup for the site reported in CfT: https://mc.a8c.com/rewind/debugger.php?site=220627588# (you can produce the backup by clicking on `Create backup link` and download it after)
* Pull the changes from this branch
* Set this [timeout to a shorter period of time](https://github.com/Automattic/studio/blob/196681da0903ae93bc9b04e548b86178e9d1808d/src/lib/wp-cli-process.ts#L14) e.g. 1 minute so that you do not have to wait for a long time for the import failure
*  Run `npm install && npm start`
* Open the Studio app
* Navigate to the Import / Export tab 
* Open `Activity Monitor` app on macOS
* Observe low CPU usage by Studio under the `CPU` tab (see screenshot for reference of what to look for):

Screenshot 2024-12-16 at 11 29 17 AM

* Drop the backup into the Import field to start the import
* Observe that the CPU usage rises drastically
* Wait for the import to fail and click on `OK` once you see the error
* Check the `Activity Monitor` and confirm that CPU usage is dropped back down to initial state before import (approximately) 
* This will confirm that the import process is stopped and not running any further

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Comment by katinthehatsite** [comment]
I tried both `180000` (3 minutes > increased by 1 minute from the current value) and `240000`(4 mins > increased by 2 minutes from the current value) and both worked as expected for me so I stuck with the shorter timeout. Happy to change it to a different value though if there are other suggestions.

**Comment by wojtekn** [comment]
@katinthehatsite altough it makes sense to increase the timeout, should we also ensure that when timeout is reached, the action that experiences it fails? 

It seems to be related or similar to https://github.com/Automattic/studio/issues/544 where the import process fails in Studio, but the database import continues running in the background.

File: src/lib/wp-cli-process.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export type MessageName = 'execute';
 export type WpCliResult = ReturnType< typeof executeWPCli >;
 export type MessageCanceled = { error: Error; canceled: boolean };
 
-const DEFAULT_RESPONSE_TIMEOUT = 120000;
+const DEFAULT_RESPONSE_TIMEOUT = 180000;
```
Comment (wojtekn) [inline src/lib/wp-cli-process.ts]: Could we make it more readable? For example:

```suggestion
const DEFAULT_RESPONSE_TIMEOUT_MS = 180 * 1000;
```

Comment (katinthehatsite) [reply src/lib/wp-cli-process.ts]: Updated in https://github.com/Automattic/studio/pull/741/commits/b9cd537ede2e40e8fde2177de44ecd9263056d16

File: src/lib/wp-cli-process.ts [hunk 2]
```diff
@@ -122,8 +122,13 @@ export default class WpCliProcess {
 			};
 
 			const timeoutHandler = () => {
-				reject( new Error( `Request for message ${ originalMessage } timed out` ) );
+				// Kill the process on timeout
+				this.#killProcess().catch( ( error ) => {
+					console.error( 'Failed to kill process after timeout:', error );
+					Sentry.captureException( error );
```
Comment (katinthehatsite) [inline src/lib/wp-cli-process.ts]: Not sure we necessarily need to pass this to Sentry but could be useful to debug other errors. Happy to remove this though.

**Comment by katinthehatsite** [comment]
> altough it makes sense to increase the timeout, should we also ensure that when timeout is reached, the action that experiences it fails?
It seems to be related or similar to https://github.com/Automattic/studio/issues/544 where the import process fails in Studio, but the database import continues running in the background.

Thanks for the review @wojtekn ! I added some handling for that and updated testing steps. Let me know what you think

File: src/lib/wp-cli-process.ts (hunk 1 above)
Comment (wojtekn) [reply src/lib/wp-cli-process.ts]: Thanks! Do you see anything against adding `_MS`? We follow that practice in Calypso, too.

File: src/lib/wp-cli-process.ts (hunk 2 above)
Comment (wojtekn) [reply src/lib/wp-cli-process.ts]: Let's skip both log calls here.

**Review by wojtekn — COMMENTED** [review]
The code change looks correct. It kills the process and informs user about failed import when timeout is reached.

However, three minutes is still low as database dumps can take minutes to import. In my case, it couldn't import linked site's database within three minutes.

Let's go with 5 minutes for now, and later, we should benchmark it to see what should be the reasonable limit that allows importing bigger sites.

**Comment by katinthehatsite** [comment]
> However, three minutes is still low as database dumps can take minutes to import. In my case, it couldn't import linked site's database within three minutes.
Let's go with 5 minutes for now, and later, we should benchmark it to see what should be the reasonable limit that allows importing bigger sites.

Cool, sounds good. The timeframe of 3 minutes worked for me but I was not sure what would be the optimal amount for different environments where users might run Studio. I will adjust to five minutes 👍

File: src/lib/wp-cli-process.ts (hunk 2 above)
Comment (katinthehatsite) [reply src/lib/wp-cli-process.ts]: Adjusted in https://github.com/Automattic/studio/pull/741/commits

**Comment by katinthehatsite** [comment]
@wojtekn I made suggested changes. Let me know what you think

File: src/lib/wp-cli-process.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export type MessageName = 'execute';
 export type WpCliResult = ReturnType< typeof executeWPCli >;
 export type MessageCanceled = { error: Error; canceled: boolean };
 
-const DEFAULT_RESPONSE_TIMEOUT = 120000;
+const DEFAULT_RESPONSE_TIMEOUT = 300 * 1000;
```
Comment (sejas) [inline src/lib/wp-cli-process.ts:14]: While increasing the general timeout for every WP-CLI command. What do you think about increasing it even more for the specific import/export commands? Not a strong opinion.

We could pass the timeout as an option in `executeWpCliCommand`

https://github.com/Automattic/studio/blob/2c27eaf3b26045eca0545ca577b3a72b56544636/src/site-server.ts#L171

And then pass the value in the import
https://github.com/Automattic/studio/blob/2c27eaf3b26045eca0545ca577b3a72b56544636/src/lib/import-export/import/importers/importer.ts#L60

and export:

https://github.com/Automattic/studio/blob/2c27eaf3b26045eca0545ca577b3a72b56544636/src/lib/import-export/export/export-database.ts#L21

**Review by sejas — APPROVED** [review]
Changes look good, and 5 minutes seems a reasonable timeout.

I left a comment, but it's not a strong opinion.

File: src/lib/wp-cli-process.ts (hunk 3 above)
Comment (katinthehatsite) [reply src/lib/wp-cli-process.ts:14]: Thanks for the review and suggestions!

I think we can leave as is for now and if we run into more issues with larger databases, we could refine the process further.

**Pull request merged**